### PR TITLE
Add setup section to CONTRIBUTING.md

### LINKED ISSUE #2301: Add development env creation steps in CONTRIBUTING.md
Currently, the [`CONTRIBUTING.md`](https://github.com/optuna/optuna/blob/master/CONTRIBUTING.md) does not have the steps to setup the development environment. Maybe we could add something like this

## Development 

To create your local development environment, do the following:

First fork the repository, then

```bash
git clone https://github.com/{Your_Username}/optuna
cd optuna
python setup.py develop
```

This is just some draft I thought of, firstly I wanted to run the idea by everyone. Should we add this?

### PR DESCRIPTION
**Pull request by himkt**
Resolve #2301.

## Motivation
This PR adds an instruction about setting up Optuna on a local computer.

## Description of the changes
- Added a section `"Setup Optuna"` to `CONTRIBUTING.md.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -35,6 +35,19 @@ If you choose to write some code, we have some conventions as follows.
...
 
+### Setup Optuna
+
+First of all, fork Optuna on GitHub.
+You can learn about fork in the official [documentation](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo).
+
+After forking, download and install Optuna on your computer.
+
+```bash
+git clone git@github.com:optuna/optuna.git
```
Comment (crcrpar) [inline CONTRIBUTING.md:46]: Why cloning the original instead of the forked one?

Comment (himkt) [reply CONTRIBUTING.md]: Sorry for my carelessness. 9a448fd fixed it.

**Review by crcrpar — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -35,6 +35,19 @@ If you choose to write some code, we have some conventions as follows.
...
 
+### Setup Optuna
+
+First of all, fork Optuna on GitHub.
+You can learn about fork in the official [documentation](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo).
+
+After forking, download and install Optuna on your computer.
+
+```bash
+git clone git@github.com:your_name/optuna.git
```
Comment (HideakiImamura) [inline CONTRIBUTING.md:46]: How about capitalizing it to make it obvious that it is a variable?
```suggestion
git clone git@github.com:YOUR_NAME/optuna.git
```

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR. I have a minor comment. PTAL.

File: CONTRIBUTING.md (hunk 2 above)
Comment (himkt) [reply CONTRIBUTING.md]: Thank you for the suggestion. I updated. 📝

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Pull request merged**